ql/src/java/org/apache/hadoop/hive/ql/exec/repl/bootstrap/ReplLoadTask.java

```diff
@@ -166,14 +185,24 @@ a database ( directory )
           }
           loadTaskTracker.update(functionsTracker);
           functionsTracker.debugLog("functions");
+
+          // Add ReplStateLogTask as function load tasks are always fully added in current cycle
+          loadTaskTracker.updateTaskCount(createFunctionReplLogTask(functionsTracker.tasks(),
+                  iterator.replLogger(), loadFunction.functionName()));
```

Comment:
The current code of ReplLoadTask is very complicated with too many Tasktrackers and iterators and some post processing happen outside getting tasks and so on. I prefer not to add more complications by moving ReplLog tasks inside those trackers. Anyways, the current way is also very clear and readable.